### PR TITLE
Update TileSets and TileMaps documentation for Godot 4.0

### PR DESCRIPTION
**Pull request by Calinou**
The documentation was split into two pages, as there are much more features to cover now.

There are still features to document, but this should be a good start to review and proofread already. I've tried to keep the documentation concise but also detailed enough to cover complex use cases, while also being accessible to beginners.

## TODO

- [x] Document terrains.
  - Documentation quality may be subpar on this aspect, as I've had trouble understanding how the terrain system works.
- [x] Document tile proxies.
- [x] Document scene tile import.

### CODE REVIEW

File: tutorials/2d/using_tilemaps.rst [hunk 1]
```diff
@@ -15,339 +21,375 @@ optimized for drawing large numbers of tiles. Finally, you can add collision,
...
+^^^^^^^^^
 
-**Template - Generic:**
+After selecting rectangle paint mode, any drawing will be performed in a
+axis-aligned rectangle.
 
-This template can be used for sideways or fully top-down perspectives.
-All templates below are designed for a TileMap cell size of 64x64, but you may
-have to use different subtile sizes for top-down templates as described below.
+If you right-click while in redctangle paint mode, erasing will be performed in
```
Comment (skyace65) [inline tutorials/2d/using_tilemaps.rst:267]:
```suggestion
If you right-click while in rectangle paint mode, erasing will be performed in
```

File: tutorials/2d/using_tilesets.rst [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+   :align: center
+   :alt: Configuring an alternative tile after clicking it in the TileSet editor
+
+   Configuring an alternative tile after clicking it in the TileSet editor
+
+After selecting an alternative tile, you can change any properties using the
+middle column like you would on a base tile. However, the list of exposed
+properties is different compared to base tiles:
+
+- **Alternative ID:** The unique numerical identifier for this alterantive tile.
```
Comment (skyace65) [inline tutorials/2d/using_tilesets.rst:487]:
```suggestion
- **Alternative ID:** The unique numerical identifier for this alternative tile.
```

**Review by skyace65 — CHANGES_REQUESTED** [review]
Didn't get to do a read through, just checked it for spelling and grammar errors. These two were the only I found.

**Comment by letmetrythisname** [comment]
Revised both documents.
[isaac-edits-diff.txt](https://github.com/godotengine/godot-docs/files/10725715/isaac-edits-diff.txt)

**Comment by Calinou** [comment]
> Great work overall. Just made a few revisions to both documents: mainly corrected uses of passive voice and truncated/split longer sentences to make them easier to read. [isaac-edits-diff.txt](https://github.com/godotengine/godot-docs/files/10725715/isaac-edits-diff.txt)


Thanks! I've applied those changes and added you as a co-author :slightly_smiling_face:

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Pushed to make CI happy (adding a newline at the end of the index file, fixing a `readadd` -> `re-add` typo and erroneous space at a beginning of a line in `using_tilemapst.rst`).

Great work, thanks!

**Pull request merged**